src/main/java/org/datadog/jmxfetch/reporter/StatsdReporter.java

```diff
@@ -50,11 +50,29 @@ public void sendServiceCheck(String checkName, String status, String message,
             init();
         }
 
-        ServiceCheck sc = new ServiceCheck(String.format("%s.can_connect", checkName),
-            this.statusToInt(status), message, hostname, tags);
+        //ActiveMQ > 5.8 hack
+        String dataName = StatsdReporter.formatDataName(String.format("%s.can_connect", checkName));
+        ServiceCheck sc = new ServiceCheck(dataName, this.statusToInt(status), message, hostname, tags);
+
         statsDClient.serviceCheck(sc);
     }
 
+    public static String formatDataName(String name){
```

Comment:
This should probably be moved to `Reporter.java` so it can be used with `ConsoleReporter` too (used by default with agent `jmx` commands).

I would also move formating logic to a new method in `Reporter.java`: it would trigger `sendServiceCheck` (private method ?) directly with the right, formatted parameters so we avoid duplicating the code among reporters.
Let me know what you think :smiley: